modules/frontend/dedicated_columns_to_json.go

```diff
@@ -0,0 +1,44 @@
+package frontend
+
+import (
+	"encoding/json"
+	"unsafe"
+
+	"github.com/grafana/tempo/tempodb/backend"
+)
+
+type DedicatedColumnsToJSON struct {
+	columnsToJSON map[uint64]string
+}
+
+func newDedicatedColumnsToJSON() *DedicatedColumnsToJSON {
+	return &DedicatedColumnsToJSON{
+		columnsToJSON: make(map[uint64]string),
+	}
+}
+
+func (d *DedicatedColumnsToJSON) JSONForDedicatedColumns(cols backend.DedicatedColumns) (string, error) {
+	if len(cols) == 0 {
+		return "", nil
+	}
+
+	hash := cols.Hash()
+	if jsonString, ok := d.columnsToJSON[hash]; ok {
+		return jsonString, nil
+	}
+
+	proto, err := cols.ToTempopb()
+	if err != nil {
```

Comment:
We can move this code into backend.DedicatedColumns:

```go
func (dcs *DedicatedColumns) MarshallJSON() ([]byte, error) {

}
```

Reply:
`MarshalJSON() ([]byte, error)` is picked up by the json encoder. I'm not sure what that would impact and I don't think we want to do that. 

https://pkg.go.dev/encoding/json#example-package-CustomMarshalJSON

The logic to go from DedicatedColumns -> Proto Objects -> Json String is weird enough I'm not sure we should put it with dedicated columns. If we moved it anywhere I think the `api` package would be appropriate. `MarshalToProtoJSON()` ?

Reply:
Makes sense to me, get everything close together

Reply:
moved dedicated columns to json thinger into `/pkg/api`